convex/game/inputs.ts

```diff
@@ -1,425 +1,240 @@
+      x: position.x,
+      y: position.y,
+      dx: facing.dx,
+      dy: facing.dy,
+      velocity: 0,
+    });
+    const playerId = await game.players.insert({
+      worldId: game.world._id,
+      name,
+      description,
+      active: true,
+      human: tokenIdentifier,
+      character,
+      locationId,
+    });
+    return playerId;
   },
-
-  // Move the player to a specified location.
-  moveTo: {
-    args: v.object({
-      playerId: v.id('players'),
-      destination: v.union(point, v.null()),
-    }),
-    returnValue: v.null(),
+} as const);
+// ...or leave, disabling the specified player.
+const leave = inputHandler({
+  args: {
+    playerId: v.id('players'),
   },
-  // Start a conversation, inviting the specified player.
-  // Conversations can only have two participants for now,
-  // so we don't have a separate "invite" input.
-  startConversation: {
-    args: v.object({
-      playerId: v.id('players'),
-      invitee: v.id('players'),
-    }),
-    returnValue: v.id('conversations'),
+  handler: async (game: AiTown, now: number, { playerId }) => {
+    const player = game.players.lookup(playerId);
+    // Stop our conversation if we're leaving the game.
+    const membership = game.conversationMembers.find((m) => m.playerId === playerId);
+    if (membership) {
+      const conversation = game.conversations.find((d) => d._id === membership.conversationId);
+      if (conversation === null) {
+        throw new Error(`Couldn't find conversation: ${membership.conversationId}`);
+      }
+      conversations.stopConversation(game, now, conversation);
+    }
+    player.active = false;
+    return null;
   },
-  // Accept an invite to a conversation, which puts the
-  // player in the "walkingOver" state until they're close
-  // enough to the other participant.
-  acceptInvite: {
-    args: v.object({
-      playerId: v.id('players'),
-      conversationId: v.id('conversations'),
-    }),
-    returnValue: v.null(),
+} as const);
+// Move the player to a specified location.
+const moveTo = inputHandler({
+  args: {
+    playerId: v.id('players'),
+    destination: v.union(point, v.null()),
   },
-  // Reject the invite. Eventually we might add a message
-  // that explains why!
-  rejectInvite: {
-    args: v.object({
-      playerId: v.id('players'),
-      conversationId: v.id('conversations'),
-    }),
-    returnValue: v.null(),
+  handler: async (game: AiTown, now: number, { playerId, destination }) => {
+    movePlayer(game, now, playerId, destination);
+    return null;
   },
-  // Leave a conversation.
-  leaveConversation: {
-    args: v.object({
-      playerId: v.id('players'),
-      conversationId: v.id('conversations'),
-    }),
-    returnValue: v.null(),
+});
+// Start a conversation, inviting the specified player.
+// Conversations can only have two participants for now,
+// so we don't have a separate "invite" input.
+const startConversation = inputHandler({
```

Comment:
might even be nice to move some of these to `conversations.ts` to match what we're doing with the agents